deeppavlov/models/nemo/asr.py

```diff
@@ -0,0 +1,193 @@
+        if has_audio:
+            max_audio_len = max(audio_lengths).item()
+
+        audio_signal = []
+        for sig, sig_len in batch:
+            if has_audio:
+                sig_len = sig_len.item()
+                if sig_len < max_audio_len:
+                    pad = (0, max_audio_len - sig_len)
+                    sig = torch.nn.functional.pad(sig, pad)
+                audio_signal.append(sig)
+
+        if has_audio:
+            audio_signal = torch.stack(audio_signal)
+            audio_lengths = torch.stack(audio_lengths)
+        else:
+            audio_signal, audio_lengths = None, None
+
+        return audio_signal, audio_lengths
+
+
+@register('nemo_asr')
+class NeMoASR(NeMoBase):
+    """ASR model on NeMo modules."""
+
+    def __init__(self, load_path: Union[str, Path], nemo_params_path: Union[str, Path], **kwargs) -> None:
+        """Initializes NeuralModules for ASR.
+
+        Args:
+            load_path: Path to a directory with pretrained checkpoints for JasperEncoder and JasperDecoderForCTC.
+            nemo_params_path: Path to a file containig labels and params for AudioToMelSpectrogramPreprocessor,
+                JasperEncoder, JasperDecoderForCTC and AudioInferDataLayer.
+
+        """
+        super(NeMoASR, self).__init__(load_path=load_path, nemo_params_path=nemo_params_path, **kwargs)
+
+        self.labels = self.nemo_params['labels']
+
+        self.data_preprocessor = AudioToMelSpectrogramPreprocessor(
+            **self.nemo_params['AudioToMelSpectrogramPreprocessor']
+        )
+        self.jasper_encoder = JasperEncoder(**self.nemo_params['JasperEncoder'])
+        self.jasper_decoder = JasperDecoderForCTC(num_classes=len(self.labels), **self.nemo_params['JasperDecoder'])
+        self.greedy_decoder = GreedyCTCDecoder()
+        self.modules_to_restore = [self.jasper_encoder, self.jasper_decoder]
+
+        self.load()
+
+    def __call__(self, audio_batch: List[Union[str, BytesIO]]) -> List[str]:
+        """Transcripts audio batch to text.
+
+        Args:
+            audio_batch: Batch to be transcribed. Elements could be either paths to audio files or Binary I/O objects.
+
+        Returns:
+            text_batch: Batch of transcripts.
+
+        """
+        data_layer = AudioInferDataLayer(audio_batch=audio_batch, **self.nemo_params['AudioToTextDataLayer'])
+        audio_signal, audio_signal_len = data_layer()
+        processed_signal, processed_signal_len = self.data_preprocessor(input_signal=audio_signal,
+                                                                        length=audio_signal_len)
+        encoded, encoded_len = self.jasper_encoder(audio_signal=processed_signal, length=processed_signal_len)
+        log_probs = self.jasper_decoder(encoder_output=encoded)
+        predictions = self.greedy_decoder(log_probs=log_probs)
+        eval_tensors = [predictions]
+        tensors = self.neural_factory.infer(tensors=eval_tensors)
```

Comment:
Is `tensors` a good namd for a variable ?

Reply:
What variable name would you use? Rename `eval_tensors` to `tensors` and `tensors` to `evaluated_tensors`?